### DISCUSSION TITLE
userEvent.clear() fails

### DISCUSSION THREAD

**Discussion by nemanjam** [testing-library/user-event#970] [Q&A]
Why simple clear() and type() in field does not work? Such a simple and trivial thing and it does not work, it creates race while typing. Very annoying and time wasting. Version: "@testing-library/user-event": "^14.2.0".

![image](https://user-images.githubusercontent.com/9990165/173227837-057c8b15-9280-436c-a0db-d83bbed1e06d.png)

Example from docs is broken.

https://testing-library.com/docs/user-event/utility#clear

![image](https://user-images.githubusercontent.com/9990165/173227876-9241b974-f0f0-43ab-af31-466343534b31.png)

![image](https://user-images.githubusercontent.com/9990165/173227988-8af1bf86-058d-4205-8cde-2778cd235354.png)

![image](https://user-images.githubusercontent.com/9990165/173242510-48a03ebf-8e51-4c63-922a-b75a1b5a9f62.png)

More info about this bug is that the input field is frozen for first 2 characters (interactions). Just typing a bit in the field fixes it. Either userEvent v14 or React Hook Form v8 are broken.

```ts
// NOTE: this fixes a bug in userEvent.clear() or React Hook Form
// field is frozen for first 2 characters
// user0 name + 123 -> user0 name3
await userEvent.type(nameInput, '123');

// edit name
await userEvent.clear(nameInput);
expect(nameInput).toHaveValue(''); // now it works
```

**Comment by fcastillo18** [comment]
Hi @nemanjam!
Did you get a solution for this? 

I'm also facing that same error after upgrading to userEvent v14

**Reply by dasm30** [reply]
Hi, I was able to make it work by calling clear twice, like so: 
```js
await userEvent.clear(nameInput);
await userEvent.clear(nameInput);
```

**Reply by ZarcoNontol** [reply]
Do we know why is this happening?

**Reply by matheustrindade** [reply]
Although it works, it seems to be just a workaround... 😞 
Do we have any update about that issue?

**Reply by kellyrmilligan** [reply]
just encountered this as well 😢  and calling it twice worked.

**Reply by yosipy** [reply]
I don't know the cause of this problem (I am using React hook form).
It may look better if you write the following

```ts
await waitFor(async () => {
  await userEvent.clear(nameInput)
  expect(nameInput).toHaveValue('')
})
```

**Reply by KovalovV** [reply]
also faced this problem
here is the part of the code that did not work correctly      
      
```
      await user.clear(firstNameInput);
      await user.clear(emailInput);
      await user.clear(lastNameInput);
      await user.clear(phoneInput);

      await user.type(firstNameInput, 'New Name');
      await user.type(lastNameInput, 'New Surname');
      await user.type(emailInput, 'new@gmail.com');
      await user.type(phoneInput, '0971111111');

      await user.click(getByText('Save'));
```

in my case it worked after I added a click event to the first input

```
      await user.click(firstNameInput);

      await user.clear(firstNameInput);
      await user.clear(emailInput);
      await user.clear(lastNameInput);
      await user.clear(phoneInput);

      await user.type(firstNameInput, 'New Name');
      await user.type(lastNameInput, 'New Surname');
      await user.type(emailInput, 'new@gmail.com');
      await user.type(phoneInput, '0971111111');
      
      await user.click(getByText('Save'));
```

**Reply by rmonteiro89** [reply]
Thanks @KovalovV! The `click` was the hint that worked for me!

**Comment by marcebdev** [comment]
This is still an issue for what it's worth, it should probably be opened as an issue because I don't see one. The additional click method worked for me though!

**Comment by josandre** [comment]
I found out that if you focus the input first, the clear works.


```javascript
 const input = screen.getByRole('input');
  input.focus();
  await userEvent.clear(input);
  await userEvent.type(input, '200')
```

**Comment by era-epoch** [comment]
None of the above solutions were working for me, so I had to force `input.value = ''` which in my case was sufficient, though not ideal.
